### PR TITLE
Update CMake to 2.8.12

### PR DESCRIPTION
**Pull request by dawagner**
And switch back to regular travis infrastructure (not container-based).

Needed by #218
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/225%23discussion_r39385343%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/225%23issuecomment-140058700%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/225%23issuecomment-140058700%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-14T12%3A43%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20400b08d7cb542a1ca95a14376b1c9cddf17e410d%20.travis.yml%2037%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/225%23discussion_r39385343%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22add%20cmake-data%22%2C%20%22created_at%22%3A%20%222015-09-14T11%3A57%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20.travis.yml%3AL10-26%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 400b08d7cb542a1ca95a14376b1c9cddf17e410d .travis.yml 37'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/225#discussion_r39385343'>File: .travis.yml:L10-26</a></b>
- <a href='https://github.com/miguelgaio'></a> add cmake-data

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/225?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/225?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/225'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/225'>Refresh</a>.<a href='#crh-update-none'></a>

File: .travis.yml [hunk 1]
```diff
@@ -10,28 +10,16 @@ compiler:
...
-            - python3-dev
-            - libasio-dev
+before_install:
+    - sudo add-apt-repository --yes ppa:ubuntu-toolchain-r/test
+    - sudo add-apt-repository --yes ppa:smspillaz/cmake-2.8.12 # Non official cmake backport
+    - sudo apt-get update -qq
 
 install:
-    - pip install --user cpp-coveralls; export PATH=$HOME/.local/bin:$PATH
+    - sudo apt-get install --yes swig cmake valgrind g++-4.8 python3-dev libasio-dev
```
Comment (miguelgaio) [inline .travis.yml]: add cmake-data

**Comment by miguelgaio** [comment]
:+1:

**Comment by krocard** [comment]
:-1: Why are you not using [ppa:kubuntu-ppa/backports](https://launchpad.net/~kubuntu-ppa/+archive/ubuntu/backports) for cmake 2.8.12.2. This ppa is [whitelisted](https://github.com/travis-ci/apt-source-whitelist/blob/master/ubuntu.json#L88)

**Comment by krocard** [comment]
Or [george-edison55-precise-backports](https://launchpad.net/~george-edison55/+archive/ubuntu/precise-backports) for cmake 3.2.3. This ppa is [whitelisted](https://github.com/travis-ci/apt-source-whitelist/blob/master/ubuntu.json#L68).

**Pull request merged**